src/librustc_mir/interpret/terminator.rs

```diff
@@ -156,10 +160,21 @@ impl<'a, 'mir, 'tcx, M: Machine<'a, 'mir, 'tcx>> InterpretCx<'a, 'mir, 'tcx, M>
                 }
             }
 
+
+            // When we encounter Resume, we've finished unwinding
+            // cleanup for the current stack frame. We pop it in order
+            // to continue unwinding the next frame
+            Resume => {
+                trace!("unwinding: resuming from cleanup");
+                assert_eq!(self.unwinding, UnwindingState::Unwinding { in_cleanup: true });
+                self.unwinding = UnwindingState::Unwinding { in_cleanup: false };
+                self.pop_stack_frame()?;
```

Comment:
This is where I'd expect to just see a parameter passed to `pop_stack_frame` saying "please use unwinding edge".

Seems like you are passing this parameter indirectly via `self.unwinding`? Why that?

Reply:
`pop_stack_frame` already needs to keep track of the current unwinding state, as it gets updated by calls into `Machine::pop_stack`.